R/http-libcurl.R

```diff
@@ -78,40 +79,24 @@ httpLibCurl <- function(protocol,
     con <- file(contentFile, "rb")
     on.exit(if (!is.null(con)) close(con), add = TRUE)
 
-    if (identical(method, "POST")) {
-      # for POST operations, send all the file's content up at once. this is necessary because some
-      # POST endpoints return 3xx status codes, which require a seekfunction in order to replay the
-      # payload (the curl package does not currently allow specifying seekfunctions from R)
-      curl::handle_setopt(handle,
-                          post = TRUE,
-                          postfields = readBin(con,
-                                               what = "raw",
-                                               n = fileLength),
-                          postfieldsize_large = fileLength)
-    } else if (identical(method, "PUT")) {
-      # for PUT operations, which are often used for larger content (bundle uploads), stream the
-      # file from disk instead of reading it from memory
-      curl::handle_setopt(handle,
-                          upload = TRUE,
-                          infilesize_large = fileLength)
-
-      curl::handle_setopt(handle,
-        readfunction = function(nbytes, ...) {
-           if (is.null(con)) {
-             return(raw())
-           }
-           bin <- readBin(con, "raw", nbytes)
-           if (length(bin) < nbytes) {
-             close(con)
-             con <<- NULL
-           }
-           bin
-        })
-    } else {
-      # why was a file specified for this endpoint?
-      warning("Content file specified, but not used because the '", method, "' request ",
-              "type does not accept a body.")
-    }
+    curl::handle_setopt(
+      handle,
+      noprogress = fileLength <= 10 * 1024^2,
```

Comment:
Show a progress bar when uploading more than 10 Mb and increase the buffer size from 64 kb to 1 Mb.

Reply:
What happens in non-interactive contexts? Is the progress output reasonable if the `rsconnect::deployApp` is captured by some CI system or otherwise sent to disk?

Reply:
I've reverted the buffer size, so it'll create a line every 64kb; so I'll make it only show when interactive.